pylint/checkers/variables.py

```diff
@@ -221,6 +221,38 @@ def _flattened_scope_names(iterator):
     return set(itertools.chain.from_iterable(values))
 
 
+def is_registered_in_singledispatch_function(node):
+    if not node.decorators:
+        return False
+
+    for decorator in node.decorators.nodes:
+        # func.register are function calls
+        if not isinstance(decorator, astroid.Call):
+            continue
+
+        func = decorator.func
+        if not isinstance(func, astroid.Attribute) or func.attrname != 'register':
+            continue
+
+        func_def = next(func.expr.infer())
+        if isinstance(func_def, astroid.FunctionDef):
+            return is_singledispatch_function(func_def)
+
+    return False
+
+
+def is_singledispatch_function(node):
```

Comment:
It's essentially reimplemented `utils.checkers.decorated_with`. I'll reuse in next commit.